apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/entity/WaTemplate.java

```diff
@@ -41,697 +41,713 @@
+        changed(command);
+    }
 
-  @OneToMany(fetch = FetchType.LAZY, mappedBy = "waTemplateUid", cascade = {
-      CascadeType.PERSIST,
-      CascadeType.MERGE,
-      CascadeType.REMOVE
-  }, orphanRemoval = true)
-  private List<WaRdbMetadatum> waRdbMetadatums;
+    public WaUiMetadata addQuestion(PageContentCommand.AddQuestion command) {
+        // Can only modify Draft pages
+        verifyDraftType();
+
+        // ensure page doesn't already contain question
+        Optional<WaUiMetadata> existing = uiMetadata.stream()
+                .filter(e -> e.getQuestionIdentifier() != null
+                        && e.getQuestionIdentifier().equals(command.question().getQuestionIdentifier()))
+                .findFirst();
+
+        if (existing.isPresent()) {
+            throw new PageContentModificationException("Unable to add a question to a page multiple times");
+        }
+
+        // Find the SubSection to add question to
+        WaUiMetadata subsection = uiMetadata.stream()
+                .filter(e -> e.getId() == command.subsection() && e.getNbsUiComponentUid() == SUB_SECTION)
+                .findFirst()
+                .orElseThrow(
+                        () -> new PageContentModificationException(
+                                "Failed to find subsection with id: " + command.subsection()));
+
+        // Questions are inserted at the END of a subsection, so find the next container (or null if subsection is at end)
+        Optional<WaUiMetadata> nextContainer = findNextElementOfComponent(
+                subsection.getOrderNbr() + 1,
+                Arrays.asList(SUB_SECTION, SECTION, TAB));
+        Integer orderNumber;
+        if (nextContainer.isEmpty()) {
+            orderNumber = uiMetadata.stream()
+                    .mapToInt(WaUiMetadata::getOrderNbr)
+                    .max()
+                    .orElseThrow(() -> new PageContentModificationException("Invalid state")) + 1;
+        } else {
+            orderNumber = nextContainer.get().getOrderNbr();
+        }
+
+        // Make room for new question
+        incrementAllFrom(orderNumber);
+
+        // Add question
+        WaUiMetadata questionEntry = new WaUiMetadata(this, command, orderNumber);
+
+        this.uiMetadata.add(questionEntry);
+
+        this.uiMetadata.sort(Comparator.comparing(WaUiMetadata::getOrderNbr));
+
+        return questionEntry;
+    }
 
-  @OneToMany(fetch = FetchType.LAZY, mappedBy = "waTemplateUid", cascade = {
-      CascadeType.PERSIST,
-      CascadeType.MERGE,
-      CascadeType.REMOVE
-  }, orphanRemoval = true)
-  private List<WaRuleMetadatum> waRuleMetadatums;
+    public void deleteQuestion(PageContentCommand.DeleteQuestion command) {
+        // Can only modify Draft pages
+        verifyDraftType();
+
+        //can not delete standard questions
+        if (command.question().isStandard()) {
+            throw new PageContentModificationException("Unable to delete standard question");
+        }
```

Comment:
Issue: I would not trust the command to have the correct `isStandard`. Instead I would perform this check on the `WaUiMetadata` entity itself